lib/apphook.c

```diff
@@ -79,14 +85,22 @@ register_application_hook(gint type, ApplicationHookFunc func, gpointer user_dat
 }
 
 static void
-run_application_hook(gint type, gboolean remove_hook)
+update_main_chain_state(gint type)
+{
+  g_assert(current_state <= type);
+  current_state = type;
+}
+
+static void
+run_application_hook(gint type)
 {
   GList *l, *l_next;
+  gboolean remove_hook = !(type & AHFLAGS_PERSISTENT);
 
   if (remove_hook)
     {
-      g_assert(current_state <= type);
-      current_state = type;
+      if (!(type & AHFLAGS_INDEPENDENT))
```

Comment:
This could be one condition, something like that:
```c
static gboolean
app_hook_should_remove(gint type)
{
    return (!(type & (AHFLAGS_PERSISTENT | AHFLAGS_INDEPENDENT)));
}
...
if (app_hook_should_remove(type))
{
  app_hook_update_state(type);
}
```

Reply:
Thanks for comment! More to it: calling update_main_chain_state should not depend on AHFLAGS_PERSISTENT. I will remove the whole if (remove_hook) around it.